tests/integration/awslambda/test_lambda.py

```diff
@@ -196,757 +129,400 @@ def read_streams(payload: T) -> T:
         create_lambda_function(
-            handler_file=TEST_LAMBDA_PYTHON_ECHO,
-            func_name=function_name,
-            runtime=LAMBDA_RUNTIME_PYTHON36,
+            func_name=func_name,
+            handler_file=lambda_fn,
+            runtime=lambda_runtime,
+            client=lambda_client,
         )
 
-        # create lambda permissions
-        action = "lambda:InvokeFunction"
-        principal = "s3.amazonaws.com"
-        statement_ids = ["s4", "s5"]
-        for sid in statement_ids:
-            resp = lambda_client.add_permission(
-                FunctionName=function_name,
-                Action=action,
-                StatementId=sid,
-                Principal=principal,
-                SourceArn=aws_stack.s3_bucket_arn("test-bucket"),
-            )
-            assert "Statement" in resp
-
-        # fetch IAM policy
-        # this is not a valid assertion in general (especially against AWS)
-        policies = iam_client.list_policies(Scope="Local", MaxItems=500)["Policies"]
-        policy_name = get_lambda_policy_name(function_name)
-        matching = [p for p in policies if p["PolicyName"] == policy_name]
-        assert 1 == len(matching)
-        assert ":policy/" in matching[0]["Arn"]
-
-        # validate both statements
-        policy = matching[0]
-        versions = iam_client.list_policy_versions(PolicyArn=policy["Arn"])["Versions"]
-        assert 1 == len(versions)
-        statements = versions[0]["Document"]["Statement"]
-        for i in range(len(statement_ids)):
-            assert action == statements[i]["Action"]
-            assert lambda_api.func_arn(function_name) == statements[i]["Resource"]
-            assert principal == statements[i]["Principal"]["Service"]
-            assert (
-                aws_stack.s3_bucket_arn("test-bucket")
-                == statements[i]["Condition"]["ArnLike"]["AWS:SourceArn"]
-            )
-            # check statement_ids in reverse order
-            assert statement_ids[abs(i - 1)] == statements[i]["Sid"]
+        first_invoke_result = lambda_client.invoke(FunctionName=func_name)
+        snapshot.match("first_invoke_result", first_invoke_result)
 
-        # remove permission that we just added
-        resp = lambda_client.remove_permission(
-            FunctionName=function_name,
-            StatementId=sid,
-            Qualifier="qual1",
-            RevisionId="r1",
-        )
-        assert 200 == resp["ResponseMetadata"]["HTTPStatusCode"]
+        second_invoke_result = lambda_client.invoke(FunctionName=func_name)
+        snapshot.match("second_invoke_result", second_invoke_result)
 
-    @pytest.mark.skip_snapshot_verify
-    @pytest.mark.snapshot
-    def test_lambda_asynchronous_invocations(
+    @pytest.mark.skip_snapshot_verify(
+        condition=is_old_provider,
+        paths=["$..FunctionError", "$..LogResult", "$..Payload", "$..Layers"],
+    )
+    @pytest.mark.skipif(is_old_provider(), reason="old provider")
```

Comment:
NIT: one of the markers can probably be removed? `skipif(is_old_provider()...)` and `skip_snapshot_verify(condtion=is_old_provider,...)`